agent/api/task.go

```diff
@@ -200,6 +207,26 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 
 }
 
+func (task *Task) addNetworkResourceProvisioningDependency() {
+	// TODO check networking mode for the task before doing this
+	for _, container := range task.Containers {
+		if container.Name == emptyHostVolumeName && container.IsInternal {
```

Comment:
Would it make more sense to change `container.IsInternal` to an iota with `normal`, `emptyHostVolume`, and `cniPause`?  Then we wouldn't have an implicit dependency on the name.